l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,289 @@
+        if (
+            _token.allowance(address(L1_SHARED_BRIDGE), address(this)) >= _amount &&
+            _token.allowance(_from, address(this)) < _amount
+        ) {
+            from = address(L1_SHARED_BRIDGE);
+        }
+        // slither-disable-next-line arbitrary-send-erc20
+        _token.safeTransferFrom(from, address(this), _amount);
+        uint256 balanceAfter = _token.balanceOf(address(this));
+
+        return balanceAfter - balanceBefore;
+    }
+
+    /// @dev Receives and parses (name, symbol, decimals) from the token contract
+    function getERC20Getters(address _token) public view returns (bytes memory) {
+        if (_token == ETH_TOKEN_ADDRESS) {
+            bytes memory name = bytes("Ether");
+            bytes memory symbol = bytes("ETH");
+            bytes memory decimals = abi.encode(uint8(18));
+            return abi.encode(name, symbol, decimals); // when depositing eth to a non-eth based chain it is an ERC20
+        }
+
+        (, bytes memory data1) = _token.staticcall(abi.encodeCall(IERC20Metadata.name, ()));
+        (, bytes memory data2) = _token.staticcall(abi.encodeCall(IERC20Metadata.symbol, ()));
+        (, bytes memory data3) = _token.staticcall(abi.encodeCall(IERC20Metadata.decimals, ()));
+        return abi.encode(data1, data2, data3);
+    }
+
+    ///  @inheritdoc IL1AssetHandler
+    function bridgeMint(
+        uint256 _chainId,
+        bytes32 _assetId,
+        bytes calldata _data
+    ) external payable override onlyBridge whenNotPaused returns (address _l1Receiver) {
+        // here we are minting the tokens after the bridgeBurn has happened on an L2, so we can assume the l1Token is not zero
+        address l1Token = tokenAddress[_assetId];
+        uint256 amount;
+        (amount, _l1Receiver) = abi.decode(_data, (uint256, address));
+        if (!L1_SHARED_BRIDGE.hyperbridgingEnabled(_chainId)) {
+            // Check that the chain has sufficient balance
+            require(chainBalance[_chainId][l1Token] >= amount, "NTV not enough funds 2"); // not enough funds
+            chainBalance[_chainId][l1Token] -= amount;
+        }
+        if (l1Token == ETH_TOKEN_ADDRESS) {
+            bool callSuccess;
+            // Low-level assembly call, to avoid any memory copying (save gas)
+            assembly {
+                callSuccess := call(gas(), _l1Receiver, amount, 0, 0, 0, 0)
+            }
+            require(callSuccess, "NTV: withdrawal failed, no funds or cannot transfer to receiver");
+        } else {
+            // Withdraw funds
+            IERC20(l1Token).safeTransfer(_l1Receiver, amount);
+        }
+        // solhint-disable-next-line func-named-parameters
+        emit BridgeMint(_chainId, _assetId, _l1Receiver, amount);
+    }
+
+    ///  @inheritdoc IL1AssetHandler
+    function bridgeRecoverFailedTransfer(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address,
```

Comment:
Why we even have this unused param? This function is only used with other protocol contracts, so it is kind of internal. No unused parameters for the future should be left.